File: superstitioApi/java/superstitioapi/shader/RingShader.java
package superstitioapi.shader;

import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.graphics.glutils.ShaderProgram;

public class RingShader {
    private static final ShaderProgram ringShader = ShaderUtility.initShader(ShaderUtility.DEFAULT_VERTEX_GLSL, "ring_fragment.glsl");
    private static final ShaderProgram ringShader_useHalfPic = ShaderUtility.initShader(ShaderUtility.DEFAULT_VERTEX_GLSL, "ring_fragment_halfPic.glsl");

    public static void setUp_ringShader(SpriteBatch sb,float radius,float halfThick,float degreeStart,float degreeLength){
        sb.setShader(ringShader);
        sb.getShader().setUniformf("u_radius", radius);
        sb.getShader().setUniformf("u_halfThick", halfThick);
        sb.getShader().setUniformf("u_degreeStart", degreeStart);
        sb.getShader().setUniformf("u_degreeLength", degreeLength);
    }

    public static void setUp_ringShader_useHalfPic(SpriteBatch sb,float radius,float halfThick,float degreeStart,float degreeLength){
        sb.setShader(ringShader_useHalfPic);
        sb.getShader().setUniformf("u_radius", radius);
        sb.getShader().setUniformf("u_halfThick", halfThick);
        sb.getShader().setUniformf("u_degreeStart", degreeStart);
        sb.getShader().setUniformf("u_degreeLength", degreeLength);
    }
}

File: superstitioApi/java/superstitioapi/shader/RingShader.kt
package superstitioapi.shader

import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.graphics.glutils.ShaderProgram

object RingShader {
    private val ringShader: ShaderProgram =
        ShaderUtility.initShader(ShaderUtility.DEFAULT_VERTEX_GLSL, "ring_fragment.glsl")
    private val ringShader_useHalfPic: ShaderProgram = ShaderUtility.initShader(
        ShaderUtility.DEFAULT_VERTEX_GLSL, "ring_fragment_halfPic" +
                ".glsl"
    )

    fun setUp_ringShader(sb: SpriteBatch, radius: Float, halfThick: Float, degreeStart: Float, degreeLength: Float) {
        sb.shader = ringShader
        sb.shader.setUniformf("u_radius", radius)
        sb.shader.setUniformf("u_halfThick", halfThick)
        sb.shader.setUniformf("u_degreeStart", degreeStart)
        sb.shader.setUniformf("u_degreeLength", degreeLength)
    }

    fun setUp_ringShader_useHalfPic(
        sb: SpriteBatch,
        radius: Float,
        halfThick: Float,
        degreeStart: Float,
        degreeLength: Float
    ) {
        sb.shader = ringShader_useHalfPic
        sb.shader.setUniformf("u_radius", radius)
        sb.shader.setUniformf("u_halfThick", halfThick)
        sb.shader.setUniformf("u_degreeStart", degreeStart)
        sb.shader.setUniformf("u_degreeLength", degreeLength)
    }
}
